### PR TITLE
TINY-6666: Moved table row type operations to snooker and updated the row dialog to use the mceTableRowType command

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-6666

Description of Changes:
* Moved the `mceTableRowType` command logic to snooker. This also included reworking snooker a little to be able to pass through the appropriate `table_header_type` logic. This is done by providing different `TableSection` instances which control how a table row/cell is transformed.
* Changed `RunOperation` in snooker to accept an object of optional behaviours to use. I've made this change to save having to pass many arguments given we're now adding one more optional argument.
* Updated the row properties dialog to use the `mceTableRowType` command

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/CHANGELOG.md [hunk 1]
```diff
@@ -63,7 +63,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Unbinding a native event handler inside the `remove` event caused an exception that blocked editor removal #TINY-7730
 - Disabled nested menu items could still be opened #TINY-7700
 - Some table operations would incorrectly cause table row attributes and styles to be lost #TINY-6666
-- The selection was incorrectly lost when using the `mceTableCellType` command #TINY-6666
+- The selection was incorrectly lost when using the `mceTableCellType` and `mceTableRowType` command #TINY-6666
+- The `mceTableRowType` was reversing the order of the rows when converting multiple header rows back to body rows #TINY-6666
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:67]: This wasn't really an intentional change, just a side-effect of moving it to snooker as it ensured the rows stayed in the correct order while doing the transformations.

**Review by ashfordneil — APPROVED** [review]

File: modules/snooker/src/main/ts/ephox/snooker/api/TableSection.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import { Fun } from '@ephox/katamari';
+import { Replication, SugarElement, SugarNode } from '@ephox/sugar';
+
+import { findTableRowHeaderType, RowHeaderType } from '../lookup/Type';
+import * as Structs from './Structs';
+import { Warehouse } from './Warehouse';
+
+type CompElm = (e1: SugarElement<Node>, e2: SugarElement<Node>) => boolean;
+type Subst = <T>(element: SugarElement<T>, comparator: CompElm) => SugarElement<T>;
```
Comment (JamesToohey) [inline modules/snooker/src/main/ts/ephox/snooker/api/TableSection.ts:9]: I'm not a huge fan of truncating words like this, can we use the full word?

Comment (JamesToohey) [reply modules/snooker/src/main/ts/ephox/snooker/api/TableSection.ts:9]: Nevermind... looks like we're already doing this elsewhere

File: modules/tinymce/CHANGELOG.md [hunk 3]
```diff
@@ -63,7 +63,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Unbinding a native event handler inside the `remove` event caused an exception that blocked editor removal #TINY-7730
 - Disabled nested menu items could still be opened #TINY-7700
 - Some table operations would incorrectly cause table row attributes and styles to be lost #TINY-6666
-- The selection was incorrectly lost when using the `mceTableCellType` command #TINY-6666
+- The selection was incorrectly lost when using the `mceTableCellType` and `mceTableRowType` command #TINY-6666
```
Comment (JamesToohey) [inline modules/tinymce/CHANGELOG.md:66]:
```suggestion
- The selection was incorrectly lost when using the `mceTableCellType` and `mceTableRowType` commands #TINY-6666
```

**Review by JamesToohey — APPROVED** [review]

**Pull request merged**